servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractTestRest.java

```diff
@@ -1852,6 +1853,62 @@ public void testBranchesHaveMetadataProperties() throws BaseNessieClientServerEx
             });
   }
 
+  @Test
+  public void testSingleBranchHasMetadataProperties() throws BaseNessieClientServerException {
+    String branchName = "singleBranchHasMetadataProperties";
+    int numCommits = 10;
+
+    Reference r = api.createReference().reference(Branch.of(branchName, null)).create();
+    String currentHash = r.getHash();
+    createCommits(r, 1, numCommits, currentHash);
+
+    // not fetching additional metadata for a single branch
+    Reference ref = api.getReference().refName(branchName).get();
+    assertThat(ref).isNotNull().isInstanceOf(Branch.class);
+    assertThat(((Branch) ref).metadataProperties()).isEmpty();
+
+    // fetching additional metadata for a single branch
+    ref = api.getReference().refName(branchName).fetchAdditionalInfo(true).get();
+    assertThat(ref).isNotNull().isInstanceOf(Branch.class);
+
+    Branch branch = (Branch) ref;
+
+    Map<String, Object> metadataProperties = branch.metadataProperties();
```

Comment:
Can this code be "merged" with the test-code from the get-references-PR?
Looks like a 1:1 duplicate

Reply:
good point, merged the code